crates/rpc-server/src/registry.rs

```diff
@@ -131,14 +135,27 @@ async fn ping() -> Result<String> {
 async fn get_block(
     Params((block_hash,)): Params<(JsonH256,)>,
     store: Data<Store>,
-) -> Result<Option<L2BlockView>> {
+    rollup_config: Data<RollupConfig>,
+) -> Result<Option<L2BlockWithStatus>> {
     let block_hash = to_h256(block_hash);
     let db = store.begin_transaction();
-    let block_opt = db.get_block(&block_hash)?.map(|block| {
-        let block_view: L2BlockView = block.into();
-        block_view
-    });
-    Ok(block_opt)
+    let block_opt = db.get_block(&block_hash)?;
+    let mut status = L2BlockStatus::Unfinalized;
+    if let Some(ref block) = block_opt {
+        if !db.reverted_block_smt()?.get(&block_hash)?.is_zero() {
+            status = L2BlockStatus::Reverted;
+        } else {
+            let tip_block_number = db.get_last_valid_tip_block()?.raw().number().unpack();
```

Comment:
We also need to check that block is exist in the block_smt & return None if it is not found.